src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -70,14 +70,20 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup>
     <!-- Enable NuGetAudit by default -->
     <NuGetAudit Condition=" '$(NuGetAudit)' == '' ">true</NuGetAudit>
-    <!-- Report on low severity vulnerabilities, and higher. Allowed values are: low, moderate, high, critical -->
+    <!-- Report all severity vulnerabilities (low severity and higher). Allowed values are: low, moderate, high, critical -->
     <NuGetAuditLevel Condition=" '$(NuGetAuditLevel)' == '' ">low</NuGetAuditLevel>
-    <!-- Report known vulnerabilities on direct dependencies only. Allowed values are: direct, all -->
+    <!-- Report known vulnerabilities on direct and transitive dependencies, unless using .NET SDK with version lower than 9.0.100. Allowed values are: direct, all -->
     <NuGetAuditMode Condition="'$(NuGetAuditMode)' == ''
                     AND '$(NuGetExeSkipSdkAnalysisLevelCheck)' != 'true'
-                    AND $([MSBuild]::VersionGreaterThanOrEquals($([MSBuild]::ValueOrDefault('$(SdkAnalysisLevel)', '0.0')), '9.0.100'))"
-                    >all</NuGetAuditMode>
-    <NuGetAuditMode Condition=" '$(NuGetAuditMode)' == '' ">direct</NuGetAuditMode>
+                    AND '$(SdkAnalysisLevel)' != ''
+                    AND $([MSBuild]::VersionLessThan('$(SdkAnalysisLevel)', '9.0.100'))"
+                    >direct</NuGetAuditMode>
+    <NuGetAuditMode Condition="'$(NuGetAuditMode)' == ''
+                    AND '$(NuGetExeSkipSdkAnalysisLevelCheck)' != 'true'
+                    AND '$(UsingMicrosoftNETSdk)' == 'true'
+                    AND '$(SdkAnalysisLevel)' == ''"
+                    >direct</NuGetAuditMode>
+    <NuGetAuditMode Condition=" '$(NuGetAuditMode)' == '' ">all</NuGetAuditMode>
```

Comment:
This is changing the nuget.exe default to all as well correct?

Can we have a test for that?

Reply:
The test I already added `NuGetAuditModeDefaults_NuGetExeEmulation` checks that the property value that nuget.exe should get will have the expected value.

What kind of test were you thinking of?

Reply:
I should add, although I added the tests to `msbuild.integration.test`, if you look at the test project, it's clear that the test is really testing NuGet.targets without any project file.  I wanted to add this test to NuGet.Build.Tasks.Tests (the unit test project), however, trying to use MSBuild's `Project` or `ProjectCollection`, the test fails because MSBuild's dlls can't be found. I tried using the MSBuild locator, but then other tests failed. It was just easier to put this test in msbuild.integration.tests.